polygeist/lib/Dialect/Polygeist/Transforms/AffineReduction.cpp

```diff
@@ -346,6 +352,24 @@ class AffineForReductionIter : public OpRewritePattern<AffineForOp> {
       });
     }
 
+    numReductionsDetected += ReductionOps.size();
+
+    [[maybe_unused]] auto getParentFunction = [](LoopLikeOpInterface loop) {
+      Operation *parentOp = loop;
+      do {
+        parentOp = parentOp->getParentOp();
+      } while (parentOp && !isa<func::FuncOp>(parentOp));
+      assert(parentOp && "Failed to find parent function");
+      return cast<func::FuncOp>(parentOp);
+    };
```

Comment:
I think we can use `getParentWithTrait` here (same in other file). We want to check for the trait instead of the type to also support `gpu::FuncOp`.

Reply:
done